video/out/opengl/hwdec_drmprime_drm.c

```diff
@@ -0,0 +1,255 @@
+        if (drm_data)
+            request = (drmModeAtomicReq *)drm_data[2];
+
+        next_frame.image = hw_image;
+        desc = (AVDRMFrameDescriptor *)hw_image->planes[0];
+
+        if (desc) {
+            int srcw = p->src.x1 - p->src.x0;
+            int srch = p->src.y1 - p->src.y0;
+            int dstw = MP_ALIGN_UP(p->dst.x1 - p->dst.x0, 2);
+            int dsth = MP_ALIGN_UP(p->dst.y1 - p->dst.y0, 2);
+
+            if (drm_prime_create_framebuffer(p->log, p->ctx->fd, desc, srcw, srch, &next_frame.fb)) {
+                ret = -1;
+                goto fail;
+            }
+
+            if (request) {
+                drm_object_set_property(request, p->ctx->overlay_plane, "FB_ID", next_frame.fb.fb_id);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "CRTC_ID", p->ctx->crtc->id);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "SRC_X",   p->src.x0 << 16);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "SRC_Y",   p->src.y0 << 16);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "SRC_W",   srcw << 16);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "SRC_H",   srch << 16);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "CRTC_X",  MP_ALIGN_DOWN(p->dst.x0, 2));
+                drm_object_set_property(request,  p->ctx->overlay_plane, "CRTC_Y",  MP_ALIGN_DOWN(p->dst.y0, 2));
+                drm_object_set_property(request,  p->ctx->overlay_plane, "CRTC_W",  dstw);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "CRTC_H",  dsth);
+                drm_object_set_property(request,  p->ctx->overlay_plane, "ZPOS",    0);
+            } else {
+                ret = drmModeSetPlane(p->ctx->fd, p->ctx->overlay_plane->id, p->ctx->crtc->id, next_frame.fb.fb_id, 0,
+                                      MP_ALIGN_DOWN(p->dst.x0, 2), MP_ALIGN_DOWN(p->dst.y0, 2), dstw, dsth,
+                                      p->src.x0 << 16, p->src.y0 << 16 , srcw << 16, srch << 16);
+                if (ret < 0) {
+                    MP_ERR(hw, "Failed to set the plane %d (buffer %d).\n", p->ctx->overlay_plane->id,
+                                next_frame.fb.fb_id);
+                    goto fail;
+                }
+            }
+        }
+    }
+
+    set_current_frame(hw, &next_frame);
+    return 0;
+    
+ fail:
+    drm_prime_destroy_framebuffer(p->log, p->ctx->fd, &next_frame.fb);
+    return ret;
+}
+
+static void uninit(struct ra_hwdec *hw)
+{
+    struct priv *p = hw->priv;
+
+    set_current_frame(hw, NULL);
+
+    if (p->ctx) {
+        drm_atomic_destroy_context(p->ctx);
+        p->ctx = NULL;
+    }
+}
+
+static int init(struct ra_hwdec *hw)
+{
+    struct priv *p = hw->priv;
+    int drm_overlay;
+
+    p->log = hw->log;
+
+    mp_read_option_raw(hw->global, "drm-overlay", &m_option_type_int, &drm_overlay);
```

Comment:
If the option value is not directly accessible somehow, use mp_get_config_group. video/out/gpu/spirv.c might be a good example of how to use it. (I regret starting using mp_read_option_raw anywhere, because it has even less type checking than the little the option system gives us.)